src/gui/screen_prusa_link.cpp

```diff
@@ -11,125 +12,46 @@
-
-public:
-    MI_PL_PASSWORD_LABEL()
-        : WI_LABEL_t(_(label), 0) {}
-
-protected:
-    void click(IWindowMenu &window_menu) override {
-    }
-};
-
-class MI_PL_PASSWORD_VALUE : public WI_LABEL_t {
-
-#ifdef USE_ST7789
-    constexpr static const char *const label = N_("");
-#else
-    constexpr static const char *const label = N_("Password");
-#endif
-    char passwd_buffer[PASSWD_STR_LENGTH];
-
-protected:
-    void printExtension(Rect16 extension_rect, color_t color_text, color_t color_back, ropfn raster_op) const override {
-        render_text_align(extension_rect, string_view_utf8::MakeRAM(reinterpret_cast<const uint8_t *>(passwd_buffer)), GuiDefaults::FontMenuSpecial, color_back, (IsFocused() && IsEnabled()) ? COLOR_DARK_GRAY : COLOR_SILVER, GuiDefaults::MenuPaddingItems, Align_t::RightCenter());
-    }
-    void click(IWindowMenu &window_menu) override {
-    }
-
-public:
-    void print_password(const char *passwd) {
-        snprintf(passwd_buffer, PASSWD_STR_LENGTH + 1, "%s", passwd);
-        InValidateExtension();
-    }
-    MI_PL_PASSWORD_VALUE()
-        : WI_LABEL_t(_(label), PASSWD_STR_LENGTH * GuiDefaults::FontMenuSpecial->w) {}
-};
-
-class MI_PL_USER : public WI_LABEL_t {
-    constexpr static const char *const label = N_("User");
-
-protected:
-    void printExtension(Rect16 extension_rect, color_t color_text, color_t color_back, ropfn raster_op) const override {
-        render_text_align(extension_rect, string_view_utf8::MakeRAM(reinterpret_cast<const uint8_t *>(PRUSA_LINK_USERNAME)), GuiDefaults::FontMenuSpecial, color_back, (IsFocused() && IsEnabled()) ? COLOR_DARK_GRAY : COLOR_SILVER, GuiDefaults::MenuPaddingItems, Align_t::RightCenter());
-    }
-    void click(IWindowMenu &window_menu) override {
-    }
-
-public:
-    MI_PL_USER()
-        : WI_LABEL_t(_(label), (sizeof(PRUSA_LINK_USERNAME) + 1) * GuiDefaults::FontMenuSpecial->w) {}
-};
+MI_PL_ENABLED::MI_PL_ENABLED()
+    : WI_SWITCH_OFF_ON_t(eeprom_get_ui8(EEVAR_PL_RUN),
+        string_view_utf8::MakeCPUFLASH((const uint8_t *)label), nullptr, is_enabled_t::yes, is_hidden_t::no) {}
 
-using PLMenuContainer = WinMenuContainer<MI_RETURN, MI_PL_ENABLED, MI_PL_REGENERATE_PASSWORD, MI_PL_USER,
-#ifdef USE_ST7789
-    MI_PL_PASSWORD_LABEL,
-#endif
-    MI_PL_PASSWORD_VALUE>;
+void MI_PL_ENABLED::OnChange(size_t old_index) {
+    Screens::Access()->Get()->WindowEvent(nullptr, GUI_event_t::CHILD_CLICK, (void *)(EventMask::value | this->index));
+}
 
-class ScreenMenuPrusaLink : public AddSuperWindow<screen_t> {
-    constexpr static const char *const label = N_("PRUSA LINK");
-    static constexpr ResourceId canvas_font = IDR_FNT_SPECIAL;
+MI_PL_PASSWORD_LABEL::MI_PL_PASSWORD_LABEL()
+    : WI_LABEL_t(_(label), 0) {}
 
-    PLMenuContainer container;
-    window_menu_t menu;
-    window_header_t header;
+void MI_PL_PASSWORD_VALUE::MI_PL_PASSWORD_VALUE::printExtension(Rect16 extension_rect, color_t color_text, color_t color_back, ropfn raster_op) const {
```

Comment:
`MI_PL_PASSWORD_VALUE::MI_PL_PASSWORD_VALUE::` Why is there MI_PL_PASSWORD_VALUE twice?

Reply:
bug
I don't know why it compiles

Reply:
fixed